include/boost/basic_any.hpp

```diff
@@ -0,0 +1,440 @@
+            : man(0), content()
+        {
+            create(*this, value, is_small_object<ValueType>());
+        }
+
+        basic_any(const basic_any & other)
+          : man(0), content()
+        {
+            if (other.man)
+            {
+                other.man(Copy, *this, &other, 0);
+            }
+        }
+
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
+        // Move constructor
+        basic_any(basic_any&& other) BOOST_NOEXCEPT
+          : man(other.man), content()
+        {
+            man(Move, *this, &other, 0);
+            other.man = 0;
+        }
+
+        // Perfect forwarding of ValueType
+        template<typename ValueType>
+        basic_any(ValueType&& value
+            , typename boost::disable_if<boost::is_same<basic_any&, ValueType> >::type* = 0 // disable if value has type `basic_any&`
+            , typename boost::disable_if<boost::is_const<ValueType> >::type* = 0) // disable if value has type `const ValueType&&`
+          : man(0), content()
+        {
+            content.large_value = 0;
+            create(*this, static_cast<ValueType&&>(value), is_small_object<BOOST_DEDUCED_TYPENAME boost::decay<ValueType>::type>());
+        }
+#endif
+
+        ~basic_any() BOOST_NOEXCEPT
+        {
+            if (man)
+            {
+                man(Destroy, *this, 0, 0);
+            }
+        }
+
+    public: // modifiers
+
+        basic_any & swap(basic_any & rhs) BOOST_NOEXCEPT
+        {
+            if (this != &rhs)
+            {
+                if (man && rhs.man)
+                {
+                    basic_any tmp;
+                    rhs.man(Move, tmp, &rhs, 0);
+                    man(Move, rhs, this, 0);
+                    tmp.man(Move, *this, &tmp, 0);
+                }
+                else if (man)
+                {
+                    man(Move, rhs, this, 0);
+                }
+                else if (rhs.man)
+                {
+                    rhs.man(Move, *this, &rhs, 0);
+                }
+            }
+            return *this;
+        }
+
+
+#ifdef BOOST_NO_CXX11_RVALUE_REFERENCES
+        template<typename ValueType>
+        basic_any & operator=(const ValueType & rhs)
+        {
+            basic_any(rhs).swap(*this);
+            return *this;
+        }
+
+        basic_any & operator=(basic_any rhs)
+        {
+            rhs.swap(*this);
+            return *this;
+        }
+
+#else 
+        basic_any & operator=(const basic_any& rhs)
+        {
+            basic_any(rhs).swap(*this);
+            return *this;
+        }
+
+        // move assignment
+        basic_any & operator=(basic_any&& rhs) BOOST_NOEXCEPT
+        {
+            rhs.swap(*this);
+            basic_any().swap(rhs);
+            return *this;
+        }
+
+        // Perfect forwarding of ValueType
+        template <class ValueType>
+        basic_any & operator=(ValueType&& rhs)
+        {
+            basic_any(static_cast<ValueType&&>(rhs)).swap(*this);
```

Comment:
Just use `std::move` instead of `static_cast<ValueType&&>(rhs)`

Reply:
It's not std::move analog but rather std::forward. I also used in boost::any. As I understand it's necessary to support compilers that don't implement std::forward yet but have support for Forwarding reference on the language level.